proto/spire/service/common/config/v1/config.proto

```diff
@@ -25,6 +29,22 @@ message ConfigureRequest {
 message ConfigureResponse {
 }
 
+message ValidateRequest {
+    // Required. Core SPIRE configuration.
+    CoreConfiguration core_configuration = 1;
+
+    // Required. HCL encoded plugin configuration.
+    string hcl_configuration = 2;
+}
+
+message ValidateResponse {
+    // Required. True when the plugin deems the configuration usable.
+    bool valid = 1;
+
+    // Required. Text message suitable for presentation to end user.
```

Comment:
Out of curiosity, what sort of information are you thinking a plugin would return in the valid case?

Reply:
Minimally, "Configuration valid" or something like that.  Really, it would depend on the plugin.  Might indicate more, might indicate less.  

For example, for the k8s_psat plugin, it might provide feedback that included the cluster name.  For another plugin, it might echo back the configuration text.  

There's not yet a distinct constraint on the message; but, having one alleviates the need to have variable logic on the success / failure cases, allowing the receiver to decide what to do without an if statement.

Reply:
One idea that would have practical value is for the "success" return value to contain reminders to unblock ports, etc.  Like I said, it would be plugin specific.